libs/autotagging-commons/src/main/java/org/opensearch/rule/InMemoryRuleProcessingService.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.rule;
+
+import org.opensearch.autotagging.Attribute;
+import org.opensearch.autotagging.FeatureType;
+import org.opensearch.autotagging.Rule;
+import org.opensearch.rule.attribute_extractor.AttributeExtractor;
+import org.opensearch.rule.storage.AttributeValueStore;
+import org.opensearch.rule.storage.AttributeValueStoreFactory;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.function.Supplier;
+
+/**
+ * This class is responsible for managing in-memory view of Rules and Find matching Rule for the request
+ */
+public class InMemoryRuleProcessingService {
+
+    /**
+     *  Main constructor which should be initialised
+     */
+    public InMemoryRuleProcessingService() {}
+
+    /**
+     * This methods adds the support for a feature to use auto-tagging framework
+     * @param featureType
+     * @param attributeValueStoreSupplier
+     */
+    public void enableFeatureType(FeatureType featureType, Supplier<AttributeValueStore<String, String>> attributeValueStoreSupplier) {
```

Comment:
I assume InMemoryRuleProcessingService is a singleton class shared across different feature types? As otherwise if we are creating this class per FeatureType, we can do this at the constructor level too.